### PR TITLE
395 tx more repos

### LINKED ISSUE #395: Transaction support for Repositories

### PR DESCRIPTION
**Pull request by clemens-msupply**
Add transaction support to all remaining repositories

No major changes, the most remarkable change I remember is moving one loader from the repository dir into the loader dir.

Closes #395

### CODE REVIEW

File: src/database/repository/diesel/central_sync_buffer.rs [hunk 1]
```diff
@@ -1,89 +1,80 @@
...
         use crate::database::schema::diesel_schema::central_sync_buffer::dsl::*;
         diesel::insert_into(central_sync_buffer)
             .values(central_sync_buffer_rows)
             // See https://github.com/diesel-rs/diesel/issues/1822.
-            .execute(&*(*connection))?;
+            .execute(self.connection.connection.deref())?;
         Ok(())
     }
 
+    // TODO this looks buggy, you could lose the entry when the server crashes just after this call
```
Comment (andreievg) [inline src/database/repository/diesel/central_sync_buffer.rs:64]: looks like it's only used in tests (where itself is tested)

File: src/database/repository/diesel/central_sync_cursor.rs [hunk 2]
```diff
@@ -1,41 +1,36 @@
...
-    pub fn update_cursor_tx(connection: &DBConnection, cursor: u32) -> Result<(), RepositoryError> {
         use crate::database::schema::diesel_schema::central_sync_cursor::dsl::*;
         let row = CentralSyncCursorRow { id: cursor as i32 };
-        diesel::delete(central_sync_cursor).execute(connection)?;
-        diesel::insert_into(central_sync_cursor)
-            .values(&row)
-            .execute(connection)?;
-        Ok(())
+        // note: if already in a transaction this creates a safepoint:
+        self.connection.connection.transaction(|| {
```
Comment (andreievg) [inline src/database/repository/diesel/central_sync_cursor.rs:28]: if an error happens, does it roll back to safepoint ?

Comment (clemens-msupply) [reply src/database/repository/diesel/central_sync_cursor.rs:28]: yes but if the error is not handled in the parent transaction the whole transaction will get aborted

**Comment by clemens-msupply** [comment]
Agree to avoid transactions in repos. This could/should be moved to a service.

**Pull request merged**